src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1482 @@
+        self.Manufacturer = manufacturer
+        if institution_name is not None:
+            self.InstitutionName = institution_name
+            if institutional_department_name is not None:
+                self.InstitutionalDepartmentName = institutional_department_name
+        self.DeviceSerialNumber = device_serial_number
+        self.ManufacturerModelName = manufacturer_model_name
+        self.SoftwareVersions = software_versions
+
+        # Presentation State Identification
+        _check_code_string(content_label)
+        self.ContentLabel = content_label
+        if content_description is not None and len(content_description) > 64:
+            raise ValueError(
+                'Argument "content_description" must not exceed 64 characters.'
+            )
+        self.ContentDescription = content_description
+        now = datetime.datetime.now()
+        self.PresentationCreationDate = DA(now.date())
+        self.PresentationCreationTime = TM(now.time())
+
+        # Not technically part of PR IODs, but we include anyway
+        self.ContentDate = DA(now.date())
+        self.ContentTime = TM(now.time())
+
+        if concept_name is not None:
+            if not isinstance(concept_name, (Code, CodedConcept)):
+                raise TypeError(
+                    'Argument "concept_name" should be of type '
+                    'pydicom.sr.coding.Code or '
+                    'highdicom.sr.CodedConcept.'
+                )
+            self.ConceptNameCodeSequence = [
+                CodedConcept(
+                    concept_name.value,
+                    concept_name.scheme_designator,
+                    concept_name.meaning,
+                    concept_name.scheme_version
+                )
+            ]
+
+        if content_creator_name is not None:
+            check_person_name(content_creator_name)
+        self.ContentCreatorName = content_creator_name
+
+        if content_creator_identification is not None:
+            if not isinstance(
+                content_creator_identification,
+                ContentCreatorIdentificationCodeSequence
+            ):
+                raise TypeError(
+                    'Argument "content_creator_identification" must be of type '
+                    'ContentCreatorIdentificationCodeSequence.'
+                )
+            self.ContentCreatorIdentificationCodeSequence = \
+                content_creator_identification
+
+        # Presentation State Relationship
+        ref_im_seq = []
+        for im in referenced_images:
+            ref_im_item = Dataset()
+            ref_im_item.ReferencedSOPClassUID = im.SOPClassUID
+            ref_im_item.ReferencedSOPInstanceUID = im.SOPInstanceUID
+            ref_im_seq.append(ref_im_item)
+        ref_series_item = Dataset()
+        ref_series_item.SeriesInstanceUID = ref_series_uid
+        ref_series_item.ReferencedImageSequence = ref_im_seq
+        self.ReferencedSeriesSequence = [ref_series_item]
```

Comment:
I am afraid this is not general enough to handle all Softcopy Presentation State IODs. For example, the Advanced Blending Softcopy Presentation State IOD does not have a Presentation State Relationship module, but instead has a Advanced Blending Presentation State module. In this case, the `ReferencedSeriesSequence` is included in the `AdvancedBlendingSequence` instead of the `ReferencedSeriesSequence`:
https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.33.html#para_0e6c2e84-f3d8-4f67-b93a-14c9653776a8

Reply:
Using inheritance may not be the best approach here. We could instead create functions to check and add sets of attributes (modules, macros, etc.) to a SOP instance.

Reply:
`_add_presentation_state_identification_attributes()`, `_add_graphic_annotation_attributes()`, etc.